components/blockList/index.js

```diff
@@ -0,0 +1,146 @@
+
+            var elClone = elTarget.cloneNode(true);
+
+            elClone.classList.add('hidden');
+
+            elTarget.insertAdjacentHTML('afterend', elClone.outerHTML);
+
+            setTimeout(function () {
+                var elClone2 = self.$el.querySelector('.hidden');
+
+                elClone2.addEventListener(transitionEndEventName, function () {
+                    self.appReference.duplicate(index);
+                    self.$el.removeChild(elClone2);
+                });
+
+                elClone2.classList.remove('hidden');
+            }, 200);
+        },
+        /**
+         * Move a block above or below a sibling
+         * @param  {Number} index     Index of block to move
+         * @param  {Number} steps Direction to move (-1 or 1)
+         */
+        shiftBlock: function (index, steps) {
+            var self = this;
+
+            // Error handling
+
+            if (index < 0) {
+                console.error('index can\'t be less than zero');
+                return;
+            } else if (index === 0 && steps === -1) {
+                console.error('block is already at the top');
+                return;
+            } else if (index === self.$el.children.length - 1 && steps === 1) {
+                console.error('block is already at the bottom');
+                return;
+            } else if (index >= self.$el.children.length) {
+                console.error('index is out of bounds');
+                return;
+            }
+
+            // Cache element references
+
+            var elTarget = self.$el.querySelector(
+                '.inline-editor:nth-child(' + (index + 1) + ')');
+            var elSwapTarget;
+
+            if (steps === 1) {
+                elSwapTarget = self.$el.querySelector(
+                    '.inline-editor:nth-child(' + (index + 2) + ')');
+            } else if (steps === -1) {
+                elSwapTarget = self.$el.querySelector(
+                    '.inline-editor:nth-child(' + index + ')');
+            }
+
+            // Get element heights
+
+            var elTargetHeight = getComputedStyle(elTarget).height;
+            var elSwapTargetHeight = getComputedStyle(elSwapTarget).height;
+
+            var transitionsCompleted = 0;
+
+            function animationEnd() {
+                if (transitionsCompleted === 2) {
+                    // Update model
+                    self.appReference.move(index, steps);
+                }
+            }
+
+            elTarget.addEventListener(transitionEndEventName, function () {
+                transitionsCompleted++;
+                animationEnd();
+            });
+
+            elSwapTarget.addEventListener(transitionEndEventName, function () {
+                transitionsCompleted++;
+                animationEnd();
+            });
+
+            // Move elements
+
+            if (steps === 1) {
+                elTarget.style.top = elSwapTargetHeight;
```

Comment:
Would it be possible to use `translateY` instead of `top` here? translateX/Y are much more performant